lib/AST/Decl.cpp

```diff
@@ -6486,6 +6486,12 @@ static bool requiresNewVTableEntry(const AbstractFunctionDecl *decl) {
   auto overrideInterfaceTy = selfInterfaceTy->adjustSuperclassMemberDeclType(
       base, decl, baseInterfaceTy);
 
+  if (!base->getGenericSignature()
```

Comment:
Let's omit this part entirely and continue to miscompile this case for now.

When you do make this change, you actually want getOverrideGenericSignature() instead of base->getGenericSignature(), and we also need SILGen tests that verify a vtable thunk is emitted instead of a direct reference to the derived method.

Reply:
Okay, done!